app/services/ProcessCsvService.scala

```diff
@@ -0,0 +1,186 @@
+        .via(eitherFromFunction(dataGenerator.identifyAndDefineSheetEither(_)(hc, request, messages)))
+        .via(eitherFromFunction(dataGenerator.setValidatorCsv(_)(hc, request, messages)))
+        .runWith(Sink.head)
+
+      validatorFuture.flatMap {
+        _.fold(
+          throwable => Future.successful(Left(throwable)),
+          validator => {
+            val futureListOfErrors: Future[Seq[Either[Throwable, RowValidationResults]]] = extractBodyOfRequest(source(successUpload.downloadUrl))
+              .via(eitherFromFunction(processRow(_, successUpload.name, validator)))
+              .runWith(Sink.seq[Either[Throwable, RowValidationResults]])
+
+            futureListOfErrors.map {
+              getRowsWithNumbers(_, successUpload.name)(messages)
+            }.flatMap{
+              case Right(errorsFromRow) => checkValidityOfRows(errorsFromRow, successUpload.name, file)
+              case Left(exception) => Future(Left(exception))
+            }
+          }
+        )
+      }
+    }
+
+  def processRow(rowBytes: List[ByteString], sheetName: String, validator: DataValidator): Either[Throwable, RowValidationResults] = {
+    val rowStrings = rowBytes.map(byteString => byteString.utf8String)
+    val parsedRow = parserUtil.formatDataToValidate(rowStrings, sheetName)
+    val rowIsEmpty = parserUtil.rowIsEmpty(rowStrings)
+
+    Try {
+      validator.validateRow(Row(0, getCells(parsedRow, 0)))
+    } match {
+      case Failure(e) =>
+        Logger.warn(e.toString)
+        Left(e)
+      case Success(list) => Right(RowValidationResults(list.getOrElse(List.empty), rowIsEmpty))
+    }
+  }
+
+  def checkFileType(name: String)(implicit messages: Messages): Either[Throwable, String] = if (name.endsWith(".csv")) {
+    Right(FilenameUtils.removeExtension(name))
+  } else {
+    Left(ERSFileProcessingException(
+      Messages("ers_check_csv_file.file_type_error", name)(messages),
+      Messages("ers_check_csv_file.file_type_error", name)(messages)))
+  }
+
+@tailrec
+private[services] final def processDisplayedErrors(n: Int, list: Seq[(List[ValidationError], Int)]): Seq[(List[ValidationError], Int)] = {
+  if (n == 0) list
+  else {
+    val indexOfFirstOccurrence: Int = list.indexWhere(entry => entry._1.nonEmpty && entry._1.exists(validationError => validationError.cell.row == 0))
+    if (indexOfFirstOccurrence != -1) {
+      val listOriginalReference = list(indexOfFirstOccurrence)._1
+      val entryReplacement = (listOriginalReference.map(validationError => {
+        val cellReplaced = validationError.cell.copy(row = indexOfFirstOccurrence + 1)
+        validationError.copy(cell = cellReplaced)
+      }), indexOfFirstOccurrence)
+      processDisplayedErrors(n - 1, list.updated(indexOfFirstOccurrence, entryReplacement))
+    } else list
+  }
+}
+
+  def giveRowNumbers(list: Seq[List[ValidationError]]): Seq[List[ValidationError]] = {
+    val numberOfErrorsToDisplay: Int = appConfig.errorCount
```

Comment:
Can this be renamed as it's more of a limit on the amount of errors that will be displayed, than the number of errors that will be displayed.